dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileHandler.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayDeque;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+class ExclusionFileHandler extends DefaultHandler {
+
+  private ArrayDeque<SymbolProblemMatcher> stack = new ArrayDeque<>();
```

Comment:
It is unclear what this stack does or how it works. Some comments would be helpful. 

It might be that stack is not the best data structure here. A custom class might work better.

Reply:
Added javadoc to the stack. (Let me know if you think a custom class can express this better than stack.)